File: chatgpt-sources/python/example-test-file.py
"""Style reference for future generated pytest files.

This file is a source example for ChatGPT prompts. The repository is currently
checklist-first, so this file documents the desired shape for future generated
tests rather than acting as a collected test module.
"""

import pytest


def test_builtin_dispatches_to_dunder_method():
    class Distance:
        def __init__(self, meters):
            self.meters = meters

        def __abs__(self):
            return abs(self.meters)

    assert abs(Distance(-12)) == 12


def test_index_protocol_is_not_the_same_as_int_protocol():
    class Port:
        def __index__(self):
            return 8080

        def __int__(self):
            return 1

    assert bin(Port()) == bin(8080)
    assert hex(Port()) == hex(8080)


@pytest.mark.parametrize(
    ("text", "expected"),
    [
        ("api/v1", "api-v1"),
        ("user/profile", "user-profile"),
    ],
)
def test_parametrize_small_api_examples(text, expected):
    assert text.replace("/", "-") == expected


def test_documented_exception_shape():
    with pytest.raises(ValueError):
        int("not-a-number")

File: chatgpt-sources/cpp/example-test-file.cpp
// Style reference for future generated GoogleTest files.
//
// The repository is currently checklist-first, so this file documents the
// desired shape for future generated tests rather than acting as a compiled
// test module.

#include <gtest/gtest.h>

#include <algorithm>
#include <stdexcept>
#include <string>
#include <string_view>
#include <utility>
#include <vector>

TEST(VectorExamples, PushBackAndCheckedAccess) {
    std::vector<int> values;
    values.reserve(3);

    values.push_back(10);
    values.push_back(20);

    ASSERT_EQ(values.size(), 2);
    EXPECT_GE(values.capacity(), 3);
    EXPECT_EQ(values[0], 10);
    EXPECT_EQ(values.at(1), 20);
}

TEST(StringViewExamples, ObservesTextWithoutOwningIt) {
    std::string text = "prefix:value";
    std::string_view view{text};

    view.remove_prefix(7);

    EXPECT_EQ(view, "value");
    EXPECT_EQ(text, "prefix:value");
}

TEST(AlgorithmExamples, SortOrdersIteratorRangeInPlace) {
    std::vector<int> values{4, 1, 3, 2};
    const std::vector<int> expected{1, 2, 3, 4};

    std::sort(values.begin(), values.end());

    EXPECT_EQ(values, expected);
}

TEST(StringExamples, StoiReportsInvalidTextWithStandardException) {
    EXPECT_THROW((void)std::stoi("not-a-number"), std::invalid_argument);
}

TEST(UtilityExamples, MoveMakesOwnershipTransferVisible) {
    std::vector<std::string> source;
    source.push_back("payload");

    std::vector<std::string> target = std::move(source);

    ASSERT_EQ(target.size(), 1);
    EXPECT_EQ(target.front(), "payload");
}
